astropy/modeling/functional_models.py

```diff
@@ -572,6 +572,92 @@ def fit_deriv(x, amplitude, x_0, fwhm):
         return [d_amplitude, d_x_0, d_fwhm]
 
 
+class Voigt1D(Fittable1DModel):
+    """
+    One dimensional model for the Voigt profile.
+
+    Parameters
+    ----------
+    pos : float
+        Position of the peak
+    amp_L : float
+        The Lorentzian amplitude
+    fwhm_L : float
+        The Lorentzian full width at half maximum
+    fwhm_G : float
+        The Gaussian full width at half maximum
+
+    See Also
+    --------
+    Gaussian1D, Lorentz1D
+
+    Notes
+    -----
+    Algorithm for the computation taken from
+    McLean, A. B., Mitchell, C. E. J. & Swanston, D. M. Implementation of an
+    efficient analytical approximation to the Voigt function for photoemission
+    lineshape analysis. Journal of Electron Spectroscopy and Related Phenomena
+    69, 125-132 (1994)
+    """
+
+    pos = Parameter()
+    amp_L = Parameter()
+    fwhm_L = Parameter()
+    fwhm_G = Parameter()
```

Comment:
This one is a little funny / confusing since the `Gaussian1D` model defines the gaussian width in terms of sigma instead of the FWHM.  On the other hand the `Lorentz1D` model uses FWHM, as does the definition in the reference.  So I think this is the best choice, though it does re-open the question raised in #2131 about using consistent definition of widths for various single-peaked curves...